### DISCUSSION TITLE
Show/Hide Status Line Modules based on window width. Does not work with all modules.

### DISCUSSION THREAD

**Discussion by chris4prez** [catppuccin/tmux#415] [Help] [answered]
I like to show/hide specific modules based on how wide my current window width is. It appears with some of the default modules my logic does not work. Specifically those modules where you are required to use "set -agF" vs "set -ag".

**Working logic:**
`set -ag status-right '#{?#{>=:#{window_width},101},#{E:@catppuccin_status_uptime},}'`
Shows uptime when window is 101 width or greater

Modules that work with logic:
- @catppuccin_status_uptime
- @catppuccin_status_date_time
- @catppuccin_status_application

**Non-working logic:**
`set -agF status-right '#{?#{>=:#{window_width},101},#{E:@catppuccin_status_cpu},}'`
Module never presents no matter if window is 101 or greater

Modules that do not work with logic:
(Commonality: all modules require set -agF to function)
- @catppuccin_status_host
- @catppuccin_status_primary_ip
- @catppuccin_status_cpu

Any suggestions appreciated if this is an issues with the modules, or I should perform my logic in a different manner for items where "set -agF" is required.

**Comment by dreknix** [comment]
The problem is, that for these modules the variable must be expanded in order to execute the underlying scripts.

To write the statement expanded into the status line, you can use the following statements:

``` sh
set -g status-right '#{?#{>=:#{window_width},101},'
set -agF status-right '#{E:@catppuccin_status_cpu}'
set -ag status-right ',}}'
```

This will result in the following status line:

``` sh
$ tmux show-options -g status-right
status-right "#{?#{>=:#{window_width},101},#[fg=#f9e2af]#[fg=#11111b,bg=#f9e2af] ▐#[fg=#(/home/dreknix/.local/share/tmux/plugins/tmux-cpu/scripts/cpu_fg_color.sh),bg=#(/home/dreknix/.local/share/tmux/plugins/tmux-cpu/scripts/cpu_bg_color.sh)] #(/home/dreknix/.local/share/tmux/plugins/tmux-cpu/scripts/cpu_percentage.sh)#[reverse]█#[noreverse],}}"
```

The statement is still not working since the `,` within the condition (inside the color definition) is not escaped with `#,`. Therefore, I think that this is not possible with this logic.

**Comment by dreknix** [comment, marked as answer]
After reading [tmux issue #1332](https://github.com/tmux/tmux/issues/1332) about escaping in conditions, I found this solution:

``` sh
set -g status-right '#{?#{>=:#{window_width},101},#{?,,'
set -agF status-right '#{E:@catppuccin_status_cpu}'
set -ag status-right '},}'
```

I add an additional condition, since in the else part the comma must not be escaped.

**Reply by chris4prez** [reply]
This is so close to working great. This solved the ability of show/hiding CPU/memory. But in the process it caused all the modules before cpu and memory which should always be visible to quit presenting in the status line.

- application, host, primary_ip

```
# Status Bar
set -g status-right-length 100
set -g status-left-length 100
set -g status-left '#{E:@catppuccin_status_session}'
set -g @catppuccin_window_text '#W'          # Inactive window present app name
set -g @catppuccin_window_current_text '#W'  # Active window present app name
set -g status-right '#{E:@catppuccin_status_application}'
set -agF status-right '#{@catppuccin_status_host}'
set -agF status-right '#{E:@catppuccin_status_primary_ip}'

set -g status-right '#{?#{>=:#{window_width},101},#{?,,'
set -agF status-right '#{E:@catppuccin_status_cpu}'
set -agF status-right '#{E:@catppuccin_status_memory}'
set -ag status-right '},}'

set -ag status-right '#{?#{>=:#{window_width},101},#{E:@catppuccin_status_uptime},}'

if 'test -r /sys/class/power_supply/BAT*' {
  set -agF status-right '#{E:@catppuccin_status_battery}'
}

set -g @catppuccin_date_time_text ' %h %d %l:%M %p'
set -ag status-right '#{?#{>=:#{window_width},101},#{E:@catppuccin_status_date_time},}'

```

**Reply by dreknix** [reply]
There is a `-a` missing:

``` sh
set -ag status-right '#{?#{>=:#{window_width},101},#{?,,'
```

**Reply by chris4prez** [reply]
Thanks! That was the fix.



> After reading [tmux issue #1332](https://github.com/tmux/tmux/issues/1332) about escaping in conditions, I found this solution:
> 
> ```shell
> set -ag status-right '#{?#{>=:#{window_width},101},#{?,,'
> set -agF status-right '#{E:@catppuccin_status_cpu}'
> set -ag status-right '},}'
> ```
> 
> I add an additional condition, since in the else part the comma must not be escaped.

Thanks this was the answer! with the addition of the -a in the first line
